controllers/azuremachine_controller.go

```diff
@@ -311,6 +305,10 @@ func (r *AzureMachineReconciler) reconcileNormal(ctx context.Context, machineSco
 
 			if reconcileError.IsTerminal() {
 				machineScope.Error(err, "failed to reconcile AzureMachine", "name", machineScope.Name())
+				machineScope.SetFailureReason(capierrors.CreateMachineError)
+				machineScope.SetFailureMessage(err)
+				machineScope.SetNotReady()
+				machineScope.SetVMState(infrav1.VMStateFailed)
```

Comment:
should we also set a condition here?

Reply:
the condition is already being set on line 304